dbt_platform_helper/utils/validation.py

```diff
@@ -43,320 +26,7 @@ def validate_addons(addons: dict):
         except SchemaError as ex:
             errors[addon_name] = f"Error in {addon_name}: {ex.code}"
 
-    _validate_extension_supported_versions(
-        config={"extensions": addons},
-        extension_type="redis",
-        version_key="engine",
-        get_supported_versions=RedisProvider(
-            boto3.client("elasticache")
-        ).get_supported_redis_versions,
-    )
-    _validate_extension_supported_versions(
-        config={"extensions": addons},
-        extension_type="opensearch",
-        version_key="engine",
-        get_supported_versions=OpensearchProvider(
-            boto3.client("opensearch")
-        ).get_supported_opensearch_versions,
-    )
+    ConfigValidator().validate_supported_redis_versions({"extensions": addons})
```

Comment:
Maybe better create a single ConfigValidator and run both validations from once instance